### PR TITLE
DM-33354: Use mamba to create environment.

### PR DESCRIPTION
**Pull request by ktlim**

### CODE REVIEW

File: bin/deploy [hunk 1]
```diff
@@ -204,16 +205,16 @@ main() {
       fi
       ARGS+=('--name' "$LSST_CONDA_ENV_NAME")
       for c in $LSST_CONDA_CHANNELS; do
-        ARGS+=("-c $c")
+        ARGS+=("-c$c")
```
Comment (erykoff) [inline bin/deploy:208]: Does `—channel conda-forge` work? Would look better at least.

**Review by erykoff — APPROVED** [review]
Should work but a comment.

File: bin/deploy (hunk 1 above)
Comment (erykoff) [reply bin/deploy:208]: That should be two dashes. Stupid autocorrect.

Comment (ktlim) [reply bin/deploy:208]: It does not.

**Pull request merged**